doc/source/user/how-to-io.rst

```diff
@@ -0,0 +1,328 @@
+
+:func:`numpy.save` and :func:`numpy.savez` create binary files. To **write a
+human-readable file**, use :func:`numpy.savetxt`. The array can only be 1- or
+2-dimensional, and there's no ` savetxtz` for multiple files.
+
+Large arrays
+==============================================================================
+
+See :ref:`how-to-io-large-arrays`.
+
+******************************************************************************
+Read an arbitrarily formatted binary file ("binary blob")
+******************************************************************************
+
+Use a :doc:`structured array <basics.rec>`.
+
+**Example:**
+
+The ``.wav`` file header is a 44-byte block preceding ``data_size`` bytes of the
+actual sound data::
+
+    chunk_id         "RIFF"
+    chunk_size       4-byte unsigned little-endian integer
+    format           "WAVE"
+    fmt_id           "fmt "
+    fmt_size         4-byte unsigned little-endian integer
+    audio_fmt        2-byte unsigned little-endian integer
+    num_channels     2-byte unsigned little-endian integer
+    sample_rate      4-byte unsigned little-endian integer
+    byte_rate        4-byte unsigned little-endian integer
+    block_align      2-byte unsigned little-endian integer
+    bits_per_sample  2-byte unsigned little-endian integer
+    data_id          "data"
+    data_size        4-byte unsigned little-endian integer
+
+The ``.wav`` file header as a NumPy structured dtype::
+
+    wav_header_dtype = np.dtype([
+        ("chunk_id", (bytes, 4)), # flexible-sized scalar type, item size 4
+        ("chunk_size", "<u4"),    # little-endian unsigned 32-bit integer
+        ("format", "S4"),         # 4-byte string
+        ("fmt_id", "S4"),
+        ("fmt_size", "<u4"),
+        ("audio_fmt", "<u2"),     #
+        ("num_channels", "<u2"),  # .. more of the same ...
+        ("sample_rate", "<u4"),   #
+        ("byte_rate", "<u4"),
+        ("block_align", "<u2"),
+        ("bits_per_sample", "<u2"),
+        ("data_id", "S4"),
+        ("data_size", "u4"),
+        #
+        # the sound data itself cannot be represented here:
+        # it does not have a fixed size
+       ])
+
+    data = np.fromfile(f,dtype=wave_header_dtype)
+
+Credit: Pauli Virtanen, :ref:`advanced_numpy`.
+
+.. _how-to-io-large-arrays:
+
+******************************************************************************
+Write or read large arrays
+******************************************************************************
+
+**Arrays too large to fit in memory** can be treated like ordinary in-memory
+arrays using :func:`numpy.memmap`::
+
+    import numpy as np
+    array = np.memmap("mydata/myarray.arr", mode="r", dtype=np.int16, shape=(1024, 1024))
+
+Memory-mapping functionality is also available with `numpy.load` via the
+``mmap_mode`` keyword argument.
+
+::
+
+  large_array[some_slice] = np.load('path/to/small_array', mmap_mode='r')

```

Comment:
I think it's important to note that the first of these is totally raw array data written with `np.tofile` (or at least, `.write(arr.tobytes())`), while the second is for files output by `np.save` in the numpy format (which we can link to the spec for).